frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -154,12 +158,86 @@ private IGitRepositorySelectionProvider getGitRepositorySelectionProvider() {
 
   private void subscribeToGitRepositoryFilesChanges() {
     Topic<GitRepositoryChangeListener> topic = GitRepository.GIT_REPO_CHANGE;
-    GitRepositoryChangeListener listener = repository -> queueRepositoryUpdateAndModelRefresh();
+    GitRepositoryChangeListener listener = repository -> {
+      trackCurrentBranchChange(repository);
+      queueRepositoryUpdateAndModelRefresh();
+    };
+
     val messageBusConnection = project.getMessageBus().connect();
     messageBusConnection.subscribe(topic, listener);
     Disposer.register(project, messageBusConnection);
   }
 
+  private Notification getNotificationAboutNonManagedBranch(String branchName) {
+    val notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(
```

Comment:
Actually... what's the lifetime of the notification? I've tried it now, it disappeared pretty quickly ☹️

Reply:
Although it sticks around even after `git machete add` for the branch in question is executed from the console... but you're surely noticed that yourself anyway

Reply:
for now it expires on the repo change topic only. changes to the file from the outside and IDE must be handled separately and are not included yet